internal/jptrace/aggregator.go

```diff
@@ -0,0 +1,40 @@
+package jptrace
+
+import (
+	"iter"
+
+	"go.opentelemetry.io/collector/pdata/pcommon"
+	"go.opentelemetry.io/collector/pdata/ptrace"
+)
+
+func AggregateTraces(tracesSeq iter.Seq[[]ptrace.Traces]) iter.Seq[ptrace.Traces] {
+	return func(yield func(trace ptrace.Traces) bool) {
+		currentTrace := ptrace.NewTraces()
+		currentTraceID := pcommon.NewTraceIDEmpty()
+
+		tracesSeq(func(traces []ptrace.Traces) bool {
+			for _, trace := range traces {
+				resources := trace.ResourceSpans()
+				// TODO: add non-empty requirement to contract
+				traceID := resources.At(0).ScopeSpans().At(0).Spans().At(0).TraceID()
+				if currentTraceID == pcommon.NewTraceIDEmpty() {
+					currentTrace = trace
+					currentTraceID = traceID
```

Comment:
I don't think this clause is needed, you can go directly to comparison because on the 1st iteration it will fall through to `else` part